File: src/main/java/de/twometer/amongus/gui/EjectPage.java
package de.twometer.amongus.gui;

import de.twometer.amongus.core.AmongUs;
import de.twometer.amongus.model.GameState;
import de.twometer.amongus.model.PlayerRole;
import de.twometer.amongus.net.NetMessage;

public class EjectPage extends BasePage {

    private final int player;
    private final NetMessage.OnPlayerEjected.Result result;

    public EjectPage(int player, NetMessage.OnPlayerEjected.Result result) {
        super("Eject.html");
        this.player = player;
        this.result = result;
    }

    @Override
    public void onDomReady() {
        super.onDomReady();

        amongUs.getScheduler().runLater(1500, () -> {
            var impostors = amongUs.getSession().getPlayers().stream().filter(p -> p.getRole() == PlayerRole.Impostor && p.alive).count();
            var remainMsg = impostors == 1 ? " Impostor remains" : " Impostors remain";
            var remaining = amongUs.getSession().getConfig().isConfirmEjects() ? impostors + remainMsg : "";

            switch (result) {
                case Tie:
                    context.call("drawEjectMessage", "No one was ejected (Tie)", remaining);
                    break;
                case Skip:
                    context.call("drawEjectMessage", "No one was ejected (Skipped)", remaining);
                    break;
                case Eject:
                    var ejectedPlayer = amongUs.getSession().getPlayer(player);
                    if (amongUs.getSession().getConfig().isConfirmEjects()) {

                        var builder = new StringBuilder();
                        builder.append(ejectedPlayer.username);
                        builder.append(" was ");
                        if (ejectedPlayer.role != PlayerRole.Impostor)
                            builder.append("not ");
                        builder.append((impostors > 1) ? "An Impostor" : "The Impostor");

                        context.call("drawEjectMessage", builder.toString(), remaining);
                    } else {
                        context.call("drawEjectMessage", ejectedPlayer.getUsername() + " was ejected.", remaining);
                    }
                    break;
            }
        });
    }

    public void close() {
        if (amongUs.getStateController().getState() == GameState.End) {
            amongUs.getGuiManager().showPage(new GameEndPage());
        } else {
            amongUs.getGuiManager().showPage(new IngamePage());
        }
    }

}

File: src/main/kotlin/de/twometer/amogus/gui/EjectPage.kt
package de.twometer.amogus.gui

import de.twometer.amogus.client.AmongUsClient
import de.twometer.amogus.client.StateManager
import de.twometer.amogus.model.*

class EjectPage(val result: EjectResult) : BasePage("Eject.html") {

    override fun onLoaded() {
        val session = AmongUsClient.session ?: return
        AmongUsClient.mainScheduler.runLater(1500) {
            val numImpostors = session.players.count { it.role == PlayerRole.Impostor && it.state == PlayerState.Alive }
            var impostorMessage = if (numImpostors == 1) "1 Impostor remains" else "$numImpostors Impostors remain"
            if (!session.config.confirmEjects) impostorMessage = ""

            when (result.type) {
                EjectResultType.Tie -> call("drawEjectMessage", "No one was ejected (Tie)", impostorMessage)
                EjectResultType.Skipped -> call("drawEjectMessage", "No one was ejected (Skipped)", impostorMessage)
                EjectResultType.Ejected -> {
                    val ejectedPlayer = session.findPlayer(result.player)!!
                    if (session.config.confirmEjects) {
                        val builder = StringBuilder()
                        builder.append(ejectedPlayer.username)
                        builder.append(" was ")
                        if (ejectedPlayer.role !== PlayerRole.Impostor) builder.append("not ")
                        builder.append(if (numImpostors > 1) "An Impostor" else "The Impostor")
                        call("drawEjectMessage", builder.toString(), impostorMessage)
                    } else {
                        call("drawEjectMessage", "${ejectedPlayer.username} was ejected.", impostorMessage)
                    }
                }
            }
        }
    }

    override fun close() {
        if (StateManager.gameState == GameState.GameOver)
        // TODO PageManager.overwrite()
        else
            PageManager.overwrite(IngamePage())
    }
}